src/core/src/device/dummy.rs

```diff
@@ -0,0 +1,50 @@
+#![allow(missing_docs)]
+use device::{Device, Resources, Capabilities, SubmitInfo};
+use device::command::{GenericCommandBuffer};
+
+pub struct DummyDevice {
+    capabilities: Capabilities
+}
+
+#[derive(Copy, Clone, Debug, Eq, Hash, PartialEq)]
+pub enum DummyResources{}
+
+pub type Buffer         = u32;
```

Comment:
`()` would work too. Note that they are public in `gl` implementation for some reason I don't recall, but definitely not to expose it to the rest of the world. If you remove `pub` from them and everything still compiles - that would be great.